--- a/Makefile	2018-08-15 09:02:03.703293160 -1000
+++ b/Makefile	2018-08-15 09:05:34.672451835 -1000
@@ -1228,12 +1228,14 @@
 endif

 ifneq ($(CONFIG_ARCH_SUNXI),)
-ifeq ($(CONFIG_ARM64),)
-u-boot-sunxi-with-spl.bin: spl/sunxi-spl.bin u-boot.img u-boot.dtb FORCE
-	$(call if_changed,binman)
-else
+ifneq ($(CONFIG_ARM64),)
 u-boot-sunxi-with-spl.bin: spl/sunxi-spl.bin u-boot.itb FORCE
 	$(call if_changed,cat)
+else
+ifneq ($(CONFIG_BINMAN),)
+u-boot-sunxi-with-spl.bin: spl/sunxi-spl.bin u-boot.img u-boot.dtb FORCE
+	$(call if_changed,binman)
+endif
 endif
 endif
